src/structs.py

```diff
@@ -1223,12 +1223,25 @@ class GroundAtomsHoldQuery(Query):
     def cost(self) -> float:
         return len(self.ground_atoms)
 
+    def __str__(self) -> str:
```

Comment:
Where is this used? It doesn't seem to be part of the caption in the generated video. I'm confused how it's getting covered.

Reply:
uhhh I can't tell either, I was using it for debugging. should I get rid of it?

Reply:
you can keep it, but is there a reason you don't want it in your video caption?

Reply:
oh bc it's redundant with the info in the response str?

Reply:
wateva, fine to keep it

Reply:
for closure's sake: it's called as a total fluke, by this method https://github.com/Learning-and-Intelligent-Systems/predicators/blob/master/tests/test_online_learning.py#L73

Reply:
in particular the "Disallowed Query" part prints out the query lol